BrainPortal/public/javascripts/userfiles.js

```diff
@@ -401,6 +444,42 @@ $(function() {
 
     /* Show/Hide dynamic actions/menu elements according to current selection */
     (function () {
+      function launch_task() {
+        var parameters    = window.location.search.split(/\?|&/);
+        var launch_button = $(this).find('input.launch_tool');
+
+        for (var i = 0; i < parameters.length; i++) {
+          var [name, tool_id]     = parameters[i].split(/=/);
+          var userfile_checkboxes = $("input[name='file_ids[]']");
+
+          if (name != "prepare_tool_id") {
+            continue;
+          }
+
+          var name_by_tool_id = {};
+          for (let tool of document.getElementsByClassName("toolsLink")) {
+            var id                  = tool.dataset.toolId;
+            var name                = tool.childNodes[0].data;
+                name_by_tool_id[id] = name;
+          }
+          var tool_name   = name_by_tool_id[tool_id]
+
+          if (tool_name === undefined) {
+            continue;
+          }
+
+          /* Do we have some files selected to launch the task on? */
```

Comment:
Why do you do this IN THE LOOP

The loop should be just about finding the tool_id, and as soon as it's found, to `break` out of.